src/main.rs

```diff
@@ -425,6 +425,12 @@ fn remote_server_options(matches: &ArgMatches) -> Result<Option<Sources>> {
         // docker_credential crate expects the config path in the $DOCKER_CONFIG. Keep docker-config-json-path parameter for backwards compatibility
         env::set_var(DOCKER_CONFIG_ENV_VAR, docker_config_json_path);
     }
+    if let Ok(docker_config_path_str) = env::var(DOCKER_CONFIG_ENV_VAR) {
+        let docker_config_path = Path::new(&docker_config_path_str).join("config.json");
+        if !docker_config_path.as_path().exists() {
```

Comment:
```suggestion
        if !docker_config_path.as_path().try_exists() {
```

It's recommended to use `try_exist` instead of `exists`, see [the official docs](https://doc.rust-lang.org/std/path/struct.Path.html#method.exists)